static/app/components/projects/appStoreConnectContext.tsx

```diff
@@ -39,51 +37,58 @@ const Provider = ({children, project, organization}: ProviderProps) => {
   }, {});
 
   useEffect(() => {
-    fetchProjectDetails();
-  }, [project]);
-
-  useEffect(() => {
-    fetchAppStoreConnectStatusData();
-  }, [projectDetails]);
-
-  async function fetchProjectDetails() {
     if (!project || projectDetails) {
-      return;
+      return undefined;
     }
 
     if (project.symbolSources) {
       setProjectDetails(project);
-      return;
+      return undefined;
     }
 
-    try {
-      const response = await api.requestPromise(`/projects/${orgSlug}/${project.slug}/`);
-      setProjectDetails(response);
-    } catch {
-      // do nothing
-    }
-  }
+    let unmounted = false;
+
+    api
+      .requestPromise(`/projects/${organization.slug}/${project.slug}/`)
+      .then(responseProjectDetails => {
+        if (unmounted) {
+          return;
+        }
 
-  async function fetchAppStoreConnectStatusData() {
+        setProjectDetails(responseProjectDetails);
+      });
+
+    return () => {
+      unmounted = true;
```

Comment:
Instead of this unmounted logic, can we use `api.clear()` to cancel the outstanding requests?